synapse/storage/databases/main/event_federation.py

```diff
@@ -298,9 +298,13 @@ def _get_auth_chain_ids_using_cover_index_txn(
                     )
 
         # Add the initial set of chains, excluding the sequence corresponding to
-        # initial event.
+        # initial event. Ultimately, the chains dict will be what is pulled from the
+        # database, there is a chance that a sequence number here will end up being a
+        # '0', which doesn't exist. Don't bother sending that to the database query.
         for chain_id, seq_no in event_chains.items():
-            chains[chain_id] = max(seq_no - 1, chains.get(chain_id, 0))
+            max_sequence_result = max(seq_no - 1, chains.get(chain_id, 0))
+            if max_sequence_result > 0:
```

Comment:
I wonder if comparing to != 0 would be clearer?

```suggestion
            if max_sequence_result != 0:
```